### PR TITLE
[Feature][Plugin] Add plugin-spark-source-mongodb into v2

### PR DESCRIPTION
**Pull request by zixi0825**
---

## Purpose of this pull request

 This pull request add plugin-spark-source-mongodb. related: #750 

## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] Change does not need document change, or I will submit document change to https://github.com/InterestingLab/seatunnel-docs later

### CODE REVIEW

**Comment by zixi0825** [comment]
@CalvinKirs

File: pom.xml [hunk 1]
```diff
@@ -141,45 +142,45 @@
         <elasticsearch-spark.version>6.8.3</elasticsearch-spark.version>
         <clickhouse-jdbc.version>0.2</clickhouse-jdbc.version>
         <hbase-spark.version>1.0.0</hbase-spark.version>
+        <mongo-spark.version>2.2.0</mongo-spark.version>
         <commons-lang3.version>3.4</commons-lang3.version>
         <maven-assembly-plugin.version>2.4</maven-assembly-plugin.version>
+        <spark.scope>compile</spark.scope>
     </properties>
```
Comment (CalvinKirs) [inline pom.xml:149]: Why is there a change here?

Comment (zixi0825) [reply pom.xml:149]: Add a spark.scope, which is convenient for switching to "compile" during local debugging, and forgot to switch back to "provided" i will change it back.

**Comment by zixi0825** [comment]
@CalvinKirs @garyelephant @RickyHuo

**Comment by CalvinKirs** [comment]
please add docs to https://github.com/InterestingLab/seatunnel/tree/dev/docs/en/configuration/source-plugins
thanks

**Review by wuchunfu — CHANGES_REQUESTED** [review]
<build>
        <plugins>
            <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-deploy-plugin</artifactId>
            </plugin>

            <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-source-plugin</artifactId>
            </plugin>

            <plugin>
                <groupId>org.sonatype.plugins</groupId>
                <artifactId>nexus-staging-maven-plugin</artifactId>
            </plugin>

            <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-gpg-plugin</artifactId>
            </plugin>

            <plugin>
                <groupId>net.alchim31.maven</groupId>
                <artifactId>scala-maven-plugin</artifactId>
            </plugin>
        </plugins>
    </build>

**Comment by wuchunfu** [comment]
Here is no longer needed, can you update the upstream code

**Review by wuchunfu — APPROVED** [review]

**Review by CalvinKirs — APPROVED** [review]

**Pull request merged**